PR for #768: Create tearsheet component

matthewgallo opened the pull request:
Contributes to #768 

This adds the `CreateTearsheet` component and a sub component, `CreateTearsheetStep`.

#### What did you change?
`CreateTearsheet/CreateTearsheet.js`
`CreateTearsheet/CreateTearsheet.mdx`
`CreateTearsheet/CreateTearsheet.stories.js`
`CreateTearsheet/CreateTearsheet.test.js`
`CreateTearsheet/CreateTearsheetStep.js`
`CreateTearsheet/_create-tearsheet.scss`
`CreateTearsheet/_index.scss`
`CreateTearsheet/_storybook-styles.scss`
`CreateTearsheet/constants.js`
`CreateTearsheet/index.js`
`CreateTearsheet/preview-components/MultiStepTearsheet.js`
`packages/cloud-cognitive/src/components/_index.scss`
`packages/cloud-cognitive/src/components/index.js`
`package-settings.js`
#### How did you test and verify your work?
Storybook and `jest /CreateTearsheet/ --coverage`

*Note: @sydrosa and @dcwarwick I removed the ability to have custom next button text per step, as well as adding the functionality of focusing the first available input on a step change (if there isn't a focusable element in a step, the focus is set on the Next/Create button), and also not requiring `onNext` and `onRequestSubmit` to return promises, this is optional now.

davidmenendez commented:
what are `preview-components`?

matthewgallo commented:
Those are just components that are used in Storybook

lee-chase commented on packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.mdx:
```diff
@@ -0,0 +1,28 @@
+
+## Overview
+
+This component is an example
```
Is it an example?

matthewgallo replied:
Yeah I need to update the doc file here 👍

lee-chase commented on packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.test.js:
```diff
@@ -0,0 +1,319 @@
+/* eslint-disable react/prop-types */
```
No ref or additional prop test.

lee-chase commented on packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js:
```diff
@@ -0,0 +1,66 @@
+import { CREATE_TEARSHEET_STEP } from './constants';
+
+const componentName = 'CreateTearsheetStep';
+const blockClass = `${pkg.prefix}--tearsheet__step`;
```
Should the block class not reference 'create tearsheet'?

lee-chase replied:
Does this component need to accept additional props?

matthewgallo replied:
I don't think it need additional props, this is simply a wrapper component for each step. It does have props of it's own but not likely that it needs additional. And yes, I agree the block class should reflect the create tearsheet block class, something like this `${pkg.prefix}--tearsheet-create__step`

lee-chase commented on packages/cloud-cognitive/src/components/CreateTearsheet/_create-tearsheet.scss:
```diff
@@ -0,0 +1,52 @@
+@import 'carbon-components/scss/components/progress-indicator/progress-indicator';
+
+@mixin create-tearsheet {
+  $block-class: #{$pkg-prefix}--tearsheet;
```
Should this be 'create-tearsheet'?

lee-chase commented on packages/cloud-cognitive/src/components/CreateTearsheet/preview-components/MultiStepTearsheet.js:
```diff
@@ -0,0 +1,207 @@
+import { CreateTearsheet } from '../CreateTearsheet';
+import { CreateTearsheetStep } from '../CreateTearsheetStep';
+
+const blockClass = `${pkg.prefix}--tearsheet-wizard-preview`;
```
Should this reference 'multi step tearsheet'?

lee-chase reviewed (commented):
See comments

matthewgallo commented:
I've updated the `blockClass` naming in some of the components and added the missing tests in the [latest commit](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/785/commits/ab717871b60cea77129bdec7ad11f9a4ca02c92a) @lee-chase
